### ISSUE TITLE
encodeBase32Unpadded does’t work nicely with wrong-length input

### ISSUE THREAD

**Issue by nomeata** [emilypi/Base32#3]
Not sure if this is intentional, but I found it surprising:
```
>  Data.ByteString.Base32.encodeBase32 (Data.ByteString.pack [0x23,0x3F,0xF2, 0x06, 0xAB,0xCD,0x01])
"EM77EBVLZUAQ===="
>  Data.ByteString.Base32.encodeBase32Unpadded' (Data.ByteString.pack [0x23,0x3F,0xF2, 0x06, 0xAB,0xCD,0x01])
"ZUAQEBVL\201\132K\SO"
>  Data.ByteString.Base32.encodeBase32Unpadded (Data.ByteString.pack [0x23,0x3F,0xF2, 0x06, 0xAB,0xCD,0x01])
"*** Exception: Cannot decode byte '\xff': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
```
I would expect the `unpadded` versions to simply omit the `=` but otherwise return the same data.

**Comment by emilypi** [comment]
Ah, it looks like i have a bug somewhere in the nopad loop/tail. Thanks for catching this. It is definitely not intended, and I'll look into it this weekend.

**Comment by emilypi** [comment]
Fixed in #4: 

```haskell
П> Data.ByteString.Base32.encodeBase32 (Data.ByteString.pack [0x23,0x3F,0xF2, 0x06, 0xAB,0xCD,0x01])
"EM77EBVLZUAQ===="
П> Data.ByteString.Base32.encodeBase32Unpadded' (Data.ByteString.pack [0x23,0x3F,0xF2, 0x06, 0xAB,0xCD,0x01])
"EM77EBVLZUAQ"
П> Data.ByteString.Base32.encodeBase32Unpadded (Data.ByteString.pack [0x23,0x3F,0xF2, 0x06, 0xAB,0xCD,0x01])
"EM77EBVLZUAQ"
П> 
```

**Comment by nomeata** [comment]
Thanks, and yay for proptests :)

**Comment by emilypi** [comment]
Yeah, this library was a little neglected on that front. We've got 96% test coverage now, so this stuff should be caught if it comes up again (fingers crossed). I'll release a 0.2.0.0 version bringing it up to stable with the rest of the `baseN` libraries.

**Comment by emilypi** [comment]
Alright @nomeata I've put out published [base32-0.2.0.0](https://hackage.haskell.org/package/base32-0.2.0.0), which addresses all of your concerns, and brings the library up to speed with `base64` and `base16`. I've taken the liberty of improving performance in between this and the last versions, so you should see a fairly decent improvement in memory usage + speed of encoding and decoding. Please feel free to give as much feedback as needed for your use-cases - I never get to use base32 as often as I use hex and base64, so it's a little neglected in that sense 😄
